src/main/webapp/app/exam/manage/exams/exam-checklist-component/exam-edit-workingtime-dialog/exam-edit-working-time.component.ts

```diff
@@ -0,0 +1,58 @@
+import { NgbModal, NgbModalRef } from '@ng-bootstrap/ng-bootstrap';
+import { Component, EventEmitter, Input, OnDestroy, OnInit, Output } from '@angular/core';
+import { faHourglassHalf } from '@fortawesome/free-solid-svg-icons';
+import dayjs from 'dayjs/esm';
+import { Subscription, from } from 'rxjs';
+
+import { Exam } from 'app/entities/exam.model';
+import { ExamEditWorkingTimeDialogComponent } from 'app/exam/manage/exams/exam-checklist-component/exam-edit-workingtime-dialog/exam-edit-working-time-dialog.component';
+import { AlertService } from 'app/core/util/alert.service';
+
+@Component({
+    selector: 'jhi-exam-edit-working-time',
+    templateUrl: './exam-edit-working-time.component.html',
+})
+export class ExamEditWorkingTimeComponent implements OnInit, OnDestroy {
+    @Input() exam: Exam;
+    @Output() examChange = new EventEmitter<Exam>();
+
+    faHourglassHalf = faHourglassHalf;
+    workingTimeChangeAllowed = false;
+
+    private modalRef: NgbModalRef | null;
+    private intervalRef: any;
+    private subscription: Subscription;
+
+    constructor(
+        private modalService: NgbModal,
+        public alertService: AlertService,
+    ) {}
+
+    ngOnInit() {
+        this.checkWorkingTimeChangeAllowed();
+        this.intervalRef = setInterval(this.checkWorkingTimeChangeAllowed.bind(this), 1000);
```

Comment:
Do we really need to check this once every second?
Could we schedule a timeout for the next relevant date as well?
E.g.
1. Take the next relevant date = the smallest future date in [startDate, endDate - 5min]
2. Calculate the diff to the current date in milliseconds using dayjs
3. Add a timeout instead of an interval

Reply:
After discussing this during the testing session, I'm now only checking for the end date, as it's legit to extend the exam time before the exam has started.